sandy/core/cov.py

```diff
@@ -496,11 +496,107 @@ def decompose(self):
         Q, R = scipy.linalg.qr(M.T)
         return R.T
 
-    def glls(self, S, Vy):
+    @classmethod
+    def get_covariance(cls, std):
+        """
+        Construct the covariance matrix from the variance vector.
+
+        Parameters
+        ----------
+        std : `pandas.Series`
+            Variance vector.
+
+        Returns
+        -------
+        `CategoryCov`
+            Object containing the covariance matrix.
+
+        Example
+        -------
+        >>> S = pd.Series(np.array([0, 2, 3]), index=pd.Index([1, 2, 3]))
+        >>> cov = sandy.CategoryCov.get_covariance(S)
+        >>> cov
+                    1           2           3
+        1 0.00000e+00 0.00000e+00 0.00000e+00
+        2 0.00000e+00 2.00000e+00 0.00000e+00
+        3 0.00000e+00 0.00000e+00 3.00000e+00
+
+        >>> assert type(cov) is sandy.CategoryCov
+        """
+        cov = pd.DataFrame(np.diag(std.values),
+                           index=std.index, columns=std.index)
+        return cls(cov)
+
+    def GLS_sensitivity(self, S, Vy):
+        """
+        
+
+        Parameters
+        ----------
+        S : `pandas.DataFrame`
+            DESCRIPTION.
+        Vy : `pandas.Series`
+            DESCRIPTION.
+
+        Raises
+        ------
+        TypeError
+            DESCRIPTION.
+
+        Returns
+        -------
+        TYPE
+            DESCRIPTION.
+
+        """
+        # Posible errors:
+        if S.data.index.values.all() != S.data.columns.values.all():
+            raise TypeError("Sensitivity matrix has not symmetric indexes")
+        if S.index.values.all() != self.data.index.values.all():
+            raise TypeError('The object indexes and the sensitivity \
+                            indexes do not match')
+        if Vy.index.values.all() != self.data.index.values.all():
+            raise TypeError('The object indexes and the extra information\
+                            indexes do not match')
+        # GLS_sensitivity:
         Vx = self.data.values
-        V = S.T.dot(Vx.dot(S)) + Vy
-        C = Vx - Vx.dot(S.dot(np.linalg.inv(V).dot(S.T.dot(Vx))))
-        return self.__class__(C)
+        M = S.T.dot(Vx.dot(S)) + Vy

```

Comment:
if you want to make sure that `Vy` is a dataframe, independently on what was given, i suggest doing:
```
S_ = pd.Series(S)
Vy_ = pd.DataFrame(Vy)
```
Notice that I now call them `Vy_` and `S_` because I don't want to change the type of the inputs